src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,243 @@
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with sssctl"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__register__ipa(ipa: IPA, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a passkey with the IPA command
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use ipa command to register a FIDO2 key.
+        2. Check the output that contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered with IPA command.
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = (
+        ipa.user("user1")
+        .add()
+        .passkey_add_register(
+            pin=123456,
+            device=f"{moduledatadir}/umockdev.device",
+            ioctl=f"{moduledatadir}/umockdev.ioctl",
+            script=f"{testdatadir}/umockdev.script",
+        )
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with the IPA command"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_passkey__su(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication of user with LDAP, IPA, AD and Samba
+    :setup:
+        1. Add a user in LDAP, IPA, AD and Samba with passkey_mapping.
+        2. Setup SSSD client with FIDO and umockdev, start SSSD service.
+    :steps:
+        1. Check authentication of the user.
+    :expectedresults:
+        1. User authenticates successfully.
+    :customerscenario: False
+    """
+    suffix = type(provider).__name__.lower()
+
+    if suffix == "ldap":
+        client.sssd.domain["local_auth_policy"] = "only"
+
+    with open(f"{testdatadir}/passkey-mapping.{suffix}") as f:
+        provider.user("user1").add().passkey_add(f.read().strip())
+
+    client.sssd.start()
+
+    assert client.auth.su.passkey(
+        username="user1",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script.{suffix}",
+    )
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_passkey__su_fail_pin(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
+    """
+    :title: Check authentication deny of user with LDAP, IPA, AD and Samba with incorrect pin
+    :setup:
+        1. Add a LDAP, IPA, AD and Samba user with passkey_mapping.
+        2. Setup SSSD client with FIDO.
```

Comment:
I'd change this and copy the sentence that is used in the previous test case: `Setup SSSD client with FIDO and umockdev, start SSSD service.`

Reply:
Same applies to the following test cases